includes/admin/class-wcs-admin-post-types.php

```diff
@@ -244,57 +244,69 @@ public function remove_bulk_actions( $actions ) {
+	 * @return array
 	 */
-	public function print_bulk_actions_script() {
+	public function filter_bulk_actions( $actions ) {
+		// 'post_status' is used in CPT datastore, 'status' is used in HPOS datastore.
+        // phpcs:ignore WordPress.Security.NonceVerification.Recommended,WordPress.Security.ValidatedSanitizedInput.MissingUnslash,WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+		$post_status = $_GET['post_status'] ?? $_GET['status'] ?? '';
+
+		// List of actions to remove that are irrelevant for subscriptions
+		$actions_to_remove = array(
+			'edit',
+			'mark_processing',
+			'mark_on-hold',
+			'mark_completed',
+			'mark_cancelled',
+		);
 
-		$post_status = ( isset( $_GET['post_status'] ) ) ? $_GET['post_status'] : '';
+		// Remove actions that are not relevant for subscriptions.
+		$actions = array_filter(
+			$actions,
+			function( $action_key ) use ( $actions_to_remove ) {
+				return ! in_array( $action_key, $actions_to_remove, true );
+			},
+			ARRAY_FILTER_USE_KEY
+		);
 
-		if ( 'shop_subscription' !== get_post_type() || in_array( $post_status, array( 'cancelled', 'trash', 'wc-expired' ) ) ) {
-			return;
+		// If we are currently in trash, expired or cancelled listing. We don't need to add subscriptions specific actions.
+		if ( in_array( $post_status, array( 'cancelled', 'trash', 'wc-expired' ), true ) ) {
+			return $actions;
 		}
 
-		// Make it filterable in case extensions want to change this
-		$bulk_actions = apply_filters( 'woocommerce_subscription_bulk_actions', array(
-			'active'    => _x( 'Activate', 'an action on a subscription', 'woocommerce-subscriptions' ),
-			'on-hold'   => _x( 'Put on-hold', 'an action on a subscription', 'woocommerce-subscriptions' ),
-			'cancelled' => _x( 'Cancel', 'an action on a subscription', 'woocommerce-subscriptions' ),
-		) );
+		/**
+		 * Subscriptions bulk actions filter.
+		 *
+		 * This is a filterable array where the key is the action (will become query arg), and the value is a translatable
+		 * string.
+		 *
+		 * @since 1.0.0 - Moved over from WooCommerce Subscriptions prior to 4.0.0
+		 */
+		$subscriptions_actions = apply_filters(
+			'woocommerce_subscription_bulk_actions',
+			[
+				'active'    => _x( 'Activate', 'an action on a subscription', 'woocommerce-subscriptions' ),
+				'on-hold'   => _x( 'Put on-hold', 'an action on a subscription', 'woocommerce-subscriptions' ),
+				'cancelled' => _x( 'Cancel', 'an action on a subscription', 'woocommerce-subscriptions' ),
+			]
+		);
+
+		$actions = array_merge( $actions, $subscriptions_actions );
 
 		// No need to display certain bulk actions if we know all the subscriptions on the page have that status already
 		switch ( $post_status ) {
 			case 'wc-active':
-				unset( $bulk_actions['active'] );
+				unset( $actions['active'] );
 				break;
 			case 'wc-on-hold':
-				unset( $bulk_actions['on-hold'] );
+				unset( $actions['on-hold'] );
 				break;
 		}
 
-		?>
-		<script type="text/javascript">
```

Comment:
Apologies if this is obvious but do we need to load these scripts anymore? Even on older versions of WooCommerce?